commands/up.go

```diff
@@ -100,24 +123,30 @@ func upHandler(cmd *cobra.Command, args []string) error {
 
 		matcher := gitignore.NewMatcher(patterns)
 
-		if err = filepath.Walk("./", func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		yamlPath := path.Join(cwd, yamlFile)
 
-			if matcher.Match(strings.Split(path, "/"), info.IsDir()) {
-				if info.IsDir() {
-					return filepath.SkipDir
-				}
-				return nil
-			}
+		debug := os.Getenv("FAAS_DEBUG")
+		if debug == "1" {
+			fmt.Printf("[Watch] added: %s\n", yamlPath)
+		}
+		watcher.Add(yamlPath)
+
+		handlerMap := make(map[string]string)
 
-			if !info.IsDir() {
-				return watcher.Add(path)
+		for serviceName, service := range services.Functions {
+			handlerMap[serviceName] = path.Join(cwd, service.Handler)
+		}
+
+		for _, service := range services.Functions {
+			handlerPath := path.Join(cwd, service.Handler)
+
+			if err := addPath(watcher, handlerPath); err != nil {
+				return err
```

Comment:
Although, I do understand these two loops are doing two different things, but as we are gonna exit on error in second loop. We should merge these two loops.

Reply:
I don't think we can do them together because we have to exhaustively check all exact matches before looking for a fuzzy match

Reply:
You're right, that can be one loop.